### PR TITLE
libelf: fix build failure using mingw

### PR DESCRIPTION
**Pull request by joda01**
Fixed missing include which allows to use getenv

### Summary
Changes to recipe:  **lib/0.8.13**

#### Motivation
Fix build error with mingw under Windows

```
C:\J\workspace\cci_mirror_PR-127\conan-home\p\b\libelecbadf08d4c97\b\src\lib\version.c: In function 'elf_version':
C:\J\workspace\cci_mirror_PR-127\conan-home\p\b\libelecbadf08d4c97\b\src\lib\version.c:31:14: error: implicit declaration of function 'getenv' [-Wimplicit-function-declaration]
   31 |     if ((s = getenv("LIBELF_SANITY_CHECKS"))) {
      |              ^~~~~~
C:\J\workspace\cci_mirror_PR-127\conan-home\p\b\libelecbadf08d4c97\b\src\lib\version.c:21:1: note: 'getenv' is defined in header '<stdlib.h>'; this is probably fixable by adding '#include <stdlib.h>'
   20 | #include <private.h>
  +++ |+#include <stdlib.h>
   21 | 
C:\J\workspace\cci_mirror_PR-127\conan-home\p\b\libelecbadf08d4c97\b\src\lib\version.c:31:12: error: assignment to 'const char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
   31 |     if ((s = getenv("LIBELF_SANITY_CHECKS"))) {
      |            ^
C:\J\workspace\cci_mirror_PR-127\conan-home\p\b\libelecbadf08d4c97\b\src\lib\version.c:32:35: error: implicit declaration of function 'strtol'; did you mean 'strtok'? [-Wimplicit-function-declaration]
   32 |         _elf_sanity_checks = (int)strtol(s, (char**)NULL, 0);
      |                                   ^~~~~~
      |                                   strtok
mingw32-make[2]: *** [CMakeFiles\elf.dir\build.make:841: CMakeFiles/elf.dir/src/lib/version.c.obj] Error 1
```

#### Details
Added patch which adds `#include <stdlib.h>`

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by joda01** [comment]
Relates to https://github.com/conan-io/conan-center-index/issues/25275

File: recipes/libelf/all/conanfile.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def generate(self):
 
     def build(self):
         if self.settings.os == "Windows":
+            replace_in_file(self, os.path.join(self.source_folder, "lib", "version.c"), "#include <private.h>", "#include <private.h>\n#include <stdlib.h>")
```
Comment (SpaceIm) [inline recipes/libelf/all/conanfile.py:86]: Could you fix this through a patch file instead please?

Comment (joda01) [reply recipes/libelf/all/conanfile.py:86]: @SpaceIm Thank's for your response. It's my first time contributing to this repo. I'll try to provide a patch file.

**Comment by SpaceIm** [comment]
Please consider https://github.com/joda01/conan-center-index/pull/1

**Comment by uilianries** [comment]
@joda01 Thank you for your PR! Please, comment on the issue #4 to be able to run your PR in the CI as well.

File: recipes/libelf/all/conanfile.py (hunk 1 above)
Comment (joda01) [reply recipes/libelf/all/conanfile.py]: Replaced by patch file!

**Comment by AbrilRBS** [comment]
@joda01 please take a look into @SpaceIm PR, it's a better way to implement patches.

I do have a question: Has this issue been reported tp upstream? I'm sure they'd appreciate getting a heads-up!

**Comment by jcar87** [comment]
Thanks @joda01 for reporting this. I've amended the PR description with a copy of the actual compilation error, for traceability and in case its useful to others.

With your proposed fix of added the missing include, that broke the build with MSVC - but fixed it with gcc/mingw. Looking further into it, it shouldn't have been necessary to include stdlib.h - as that should've been handled by the config.h - which shouldve already been including it (via private.h). Turns out that there was a bug in the Conan Center specific CMakeLists.txt - I've fixed that and now it should work for both msvc and mingw.

I've enabled a mingw build for this PR, results here: https://github.com/conan-io/conan-center-index/pull/25276/checks?check_run_id=34040477957 - as evidence this fixes your issue :)

Thanks for contributing to Conan Center!

**Pull request merged**